File: runtime/Java/src/main/java/org/antlr/v5/runtime/java/ConsoleErrorListener.java
/*
 * Copyright (c) 2012-present The ANTLR Project. All rights reserved.
 * Use of this file is governed by the BSD 3-clause license that
 * can be found in the LICENSE.txt file in the project root.
 */
package org.antlr.v5.runtime.java;

import org.antlr.v5.runtime.core.Recognizer;
import org.antlr.v5.runtime.core.error.RecognitionException;

/**
 *
 * @author Sam Harwell
 */
public class ConsoleErrorListener extends BaseErrorListener {
	/**
	 * Provides a default instance of {@link ConsoleErrorListener}.
	 */
	public static final ConsoleErrorListener INSTANCE = new ConsoleErrorListener();

	/**
	 * {@inheritDoc}
	 *
	 * <p>
	 * This implementation prints messages to {@link System#err} containing the
	 * values of {@code line}, {@code charPositionInLine}, and {@code msg} using
	 * the following format.</p>
	 *
	 * <pre>
	 * line <em>line</em>:<em>charPositionInLine</em> <em>msg</em>
	 * </pre>
	 */
	@Override
	public void syntaxError(Recognizer<?, ?> recognizer,
							Object offendingSymbol,
							int line,
							int charPositionInLine,
							String msg,
							RecognitionException e)
	{
		System.err.println("line " + line + ":" + charPositionInLine + " " + msg);
	}

}

File: runtime/Kotlin/src/org/antlr/v5/runtime/kotlin/ConsoleErrorListener.kt
/*
 * Copyright (c) 2012-present The ANTLR Project. All rights reserved.
 * Use of this file is governed by the BSD 3-clause license that
 * can be found in the LICENSE.txt file in the project root.
 */
package org.antlr.v5.runtime.kotlin

import org.antlr.v5.runtime.core.Recognizer
import org.antlr.v5.runtime.core.error.RecognitionException

/**
 * @author Sam Harwell
 */
public open class ConsoleErrorListener : BaseErrorListener() {
  public companion object {
    /**
     * Provides a default instance of [ConsoleErrorListener].
     */
    public val INSTANCE: ConsoleErrorListener = ConsoleErrorListener()
  }

  /**
   * This implementation prints messages to [System.err] containing the
   * values of [line], [charPositionInLine], and [msg] using the following format:
   *
   * ```
   * line *line*:*charPositionInLine* *msg*
   * ```
   */
  override fun syntaxError(
      recognizer: Recognizer<*, *>,
      offendingSymbol: Any?,
      line: Int,
      charPositionInLine: Int,
      msg: String,
      e: RecognitionException?,
  ) {
    System.err.println("line $line:$charPositionInLine $msg")
  }
}
